src/main/scala/li/cil/oc/server/machine/StringBufferValue.scala

```diff
@@ -0,0 +1,100 @@
+package li.cil.oc.server.machine
+
+import li.cil.oc.{OpenComputers, Settings}
+import li.cil.oc.api.machine.{Arguments, Callback, Context}
+import li.cil.oc.api.prefab.AbstractValue
+import li.cil.oc.server.machine.luac.NativeLuaArchitecture
+import net.minecraft.nbt.NBTTagCompound
+
+
+class StringBufferValue() extends AbstractValue {
+  /* TODO: fix koefficients */
+  final val limitBase = 10000
+  final val readCost = 0.1
+  final val writeCost = 0.2
```

Comment:
Please notice, I'm not sure what is best values for this

Reply:
"koefficients", what does that mean? It's not a valid english word. :p
From what I understand, there shouldn't be a performance penalty for this, Lua already has an operation limit, and memory operations should be fast.
Memory size limit is harder, theoretically it should be limited by the memory installed in the OC computer?

Reply:
Oh, sorry, that should be 'factor' or something like this.
Memory size limit is already bound to computer RAM, and when buffer is allocated max lua memory is being substracted by allocated memory size, so there's no need for additional setup for that.
And yes, this setups costs (limits) for buffer I/O. Keeping in mind that it should be fast it now consumes `(opCost * size + 1) / limitBase` of call budget per call to read/write/tostring and `1 / (limitBase * 10)` for length operation, where call budget is `1.0` for Tier 2 computer; `opCost` is `readCost` or `writeCost`; `size` is requested (not actual) operation size. But I think that this should be tuned...

Reply:
I guess I should use tables unless I need very big buffers?

Reply:
Yes, you won't win that much If you'll use it on small buffers

Reply:
You mean I'll always get a negative performance "boost" at a higher data density? Why can't we have both?

Reply:
Well, if API would be implemented as 'raw' functions, not userdata callbacks then you can win some cpu time. Lua arrays are O(logN) for read/write and when N is small it's close enough to O(1), so you will lose more on calling userdata callback, unwrapping parametes, then wrapping results, etc. But in fact I wrote this API not for speed, but memory consumption optimization - here you should get improvement even on small arrays.